apisix/plugins/log-rotate.lua

```diff
@@ -219,18 +209,67 @@ local function init_default_logs(logs_info, log_type)
 end
 
 
+local function file_size(file)
+    local attr = lfs.attributes(file)
+    if attr then
+        return attr.size
+    end
+    return 0
+end
+
+
+local function rotate_file(files, now_time, max_kept)
+    for _, file in ipairs(files) do
+        local now_date = os_date("%Y-%m-%d_%H-%M-%S", now_time)
+        local new_file = rename_file(default_logs[file], now_date)
+        if not new_file then
+            return
+        end
+
+        local pid = process.get_master_pid()
+        local sig_user1 = signal.signum("USR1")
+
+        core.log.warn("send USR1 signal to master process [", pid, "] for reopening log file")
+
+        if (pid and sig_user1) then
```

Comment:
I am curious in what situation Nginx doesn't have master pid.
And the SIGUSR1 is part of the POSIX standard. APISIX requires to be run on a platform that is POSIX compatible.

Reply:
Same as signal.signum, get_master_pid's return value maybe null. If we use the return value directly, a warn is reported with Lua plugin in vscode.
If this is a nonsense warning message, i'll revert this change.

Reply:
https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/process.md#get_master_pid
`get_master_pid` returns nil when it is running on Nginx < 1.13.8. So the check is unnecessary in our case.

Reply:
fixed